### ISSUE TITLE
Automatically change the Class Name when changing its file name

### ISSUE THREAD

**Issue by AllanOricil** [forcedotcom/salesforcedx-vscode#3574]
**Is your feature request related to a problem? Please describe.**
No, it will just remove a waste of time and improve DX

**Describe the solution you'd like**
When I rename a class file, its class name has to to be changed manually. Could you please automate this as well? 

**Describe alternatives you've considered**
After renaming the file I have to open it and also rename the class

**Additional context**
N/A

**Comment by randi274** [comment]
Hi @AllanOricil - have you tried using the `Rename Symbol` option from the Class Name itself? For me, this renames the class name, the file, the meta file, and references.

https://user-images.githubusercontent.com/9795193/200684124-cfa41df4-64d4-494e-97bd-f25a75e6c75c.mov

**Comment by AllanOricil** [comment]
No, I have not. Good to know about this option

**Comment by AllanOricil** [comment]
Hi @randi274 I tested this built in command and discovered that there are a few places where it does not work.

https://user-images.githubusercontent.com/55927613/200871384-742dcf60-7460-4a5a-a40e-351cfc923808.mov

Vscode does not throw exceptions explaining the reason.

**Comment by AllanOricil** [comment]
@randi274 there is really no way to isolate the root cause. I was able to rename that class from the video after doing the following:

1. tried to rename `public inherited sharing class EXP_PermissionSetSelector extends COM_PermissionSetSelector` using Rename Symbol and it did not work
2. then I removed `extends COM_PermissionSetSelector`
3. after that I tried the Rename Symbol command again, but this time it worked
4. then I added `extends COM_PermissionSetSelector` back
5. finally I used the Rename Symbol command again and it worked!

I was expecting step 5 to fail.
